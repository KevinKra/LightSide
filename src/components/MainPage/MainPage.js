import React, { Component } from "react";
// import { Link, animateScroll as scroll } from "react-scroll";
// import uuidv1 from "uuid/v1";
// import * as apiCalls from "../../utils/api/apiCalls";
import uuidv1 from "uuid/v1";
// import * as mockData from "../../utils/data";
import "./MainPage.scss";
// import Crawl from "../Crawl/Crawl";
import ContentSection from "../ContentSection/ContentSection";
import HeroSection from "../HeroSection/HeroSection";

export default class MainPage extends Component {
  state = {
    theme: "people",
    content: [{ name: "steve", gender: "dog" }],
    people: [],
    planets: [],
    vehicles: [],
    favorites: []
  };

  // componentDidMount() {
  //   scroll.scrollToTop();
  // }

  fetchData = async data => {
    const response = await fetch(`https://swapi.co/api/${data}`);
    const parse = await response.json();
    const output = parse.results.map(element => {
      return { ...element, favorited: false, id: uuidv1() };
    });
    // console.log("output", output);
    return output;
  };

  addToFavorites = target => {
    const match = this.state.content.filter(element => {
      return element.name === target;
    });
    console.log(match[0]);
    const newElement = match[0];
    newElement.favorited = true;
    const detection = this.state.favorites.filter(
      element => element.name === newElement.name
    );
    if (detection.length < 1) {
      this.setState({ favorites: [...this.state.favorites, newElement] });
    }
  };

  removeFromFavorites = target => {
    const element = this.state.favorites.find(element => {
      return (element.name = target);
    });
    element.favorited = false;
    console.log("element", element);
    const updatedFavorites = this.state.favorites.filter(element => {
      return element.name !== target;
    });
    {
      this.state.theme === "favorites" &&
        this.setState({
          favorites: updatedFavorites,
          content: updatedFavorites
        });
    }
    this.setState({ favorites: updatedFavorites });
  };

  displayData = async type => {
    this.setState({ content: [] });
    if (this.state[type].length > 1) {
      // console.log("local state");
      this.setState({ content: this.state[type], theme: type });
    } else {
      // console.log("fetching");
      this.setState({ theme: type });
      const response = await this.fetchData(type);
      if (this.state.theme === type) {
        console.log("theme", this.state.theme);
        console.log("type", type);
        this.setState({ content: response, [type]: response, theme: type });
      } else {
        console.log("theme", this.state.theme);
        console.log("type", type);
        this.setState({ [type]: response });
      }
    }
  };

  displayFavorites = () => {
    this.setState({ content: this.state.favorites, theme: "favorites" });
  };

  render() {
    return (
      <main className="MainPage">
        <HeroSection
          displayData={this.displayData}
          displayFavorites={this.displayFavorites}
        />
        <ContentSection
          content={this.state.content}
          theme={this.state.theme}
          addToFavorites={this.addToFavorites}
          removeFromFavorites={this.removeFromFavorites}
        />
      </main>
    );
  }
}
