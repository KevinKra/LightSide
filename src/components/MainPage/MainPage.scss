.MainPage {
  // border: 1px solid blue;
  overflow: hidden;
  overflow-y: none;
  min-height: 100vh;
  min-width: 100vw;
  h1 {
    margin: 0;
    border: 1px solid red;
  }
  nav {
    border: 1px solid red;
  }
  .crawl {
    // border: 1px solid blue;
    background-color: lightgrey;
    height: 100vh;
    width: 100vw;
    transition: 0.5s height ease-in;
    overflow: hidden;
  }
  .hero-section {
    border: 1px solid green;
    overflow: hidden;
    height: 100vh;
    transition: 0.5s height ease-in;
    width: 100vw;
  }
  .background-image {
    // border: 1px solid red;
    height: 100vh;
    width: 100vw;
    background: url("../../utils/images/sw-stormtroopers.jpg");
    background-size: cover;
  }
  .hero-content {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    border: 1px solid blue;
    height: 90vh;
    nav {
      align-self: center;
    }
  }
  .content-section {
    border: 1px solid red;
    height: 100vh;
    width: 100vw;
  }
}
