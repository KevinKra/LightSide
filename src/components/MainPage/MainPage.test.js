import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import MainPage from "../MainPage/MainPage";

describe("<MainPage />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MainPage />);
  });
  it("should match the snapshot", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
