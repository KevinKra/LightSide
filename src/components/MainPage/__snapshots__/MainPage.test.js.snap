// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> should match the snapshot 1`] = `
<main
  className="MainPage"
>
  <HeroSection
    backgroundImage="sw-boba.jpg"
    displayData={[Function]}
    displayFavorites={[Function]}
    totalFavorites={0}
  />
  <ContentSection
    addToFavorites={[Function]}
    content={Array []}
    logError={[Function]}
    removeFromFavorites={[Function]}
    theme="people"
  />
</main>
`;
